#! /bin/sh /usr/share/dpatch/dpatch-run
## 011enforce_activaction_key.dpatch by Giuseppe Iuculano <giuseppe@iuculano.it>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Enforce activation key to be a string (Closes: #541102)

@DPATCH@
diff -urNad wordpress~/wp-login.php wordpress/wp-login.php
--- wordpress~/wp-login.php	2009-08-12 18:12:04.000000000 +0200
+++ wordpress/wp-login.php	2009-08-12 18:12:54.000000000 +0200
@@ -187,7 +187,7 @@
 
 	$key = preg_replace('/[^a-z0-9]/i', '', $key);
 
-	if ( empty( $key ) )
+	if ( empty( $key ) || !is_string( $key ) )
 		return new WP_Error('invalid_key', __('Invalid key'));
 
 	$user = $wpdb->get_row($wpdb->prepare("SELECT * FROM $wpdb->users WHERE user_activation_key = %s", $key));
